Add a GitHub section (#127)
alphagov/reliability-engineering: source/documentation/github/access-github-alphagov.md

## Access GitHub Alphagov

> People joining GDS do not automatically have access to [Alphagov][] on [GitHub][]. To grant access to Alphagov, user needs to [create a new][] or share an existing GitHub username with the team. All users are required to enable [two-form authentication][] on their GitHub accounts.

Reviewer: ```suggestion
People joining GDS do not automatically have access to the GDS GitHub organisation, [alphagov][]. To gain access:

- [Create a GitHub account][GitHub] if you don't have one already. Existing personal accounts are fine to continue using.
- Enable [two-factor authentication][] on your GitHub account.
- Give your GitHub username to your team's tech lead, or your nearest technical Senior Management Team member.
```

> It’s the individual teams’ responsibility to grant that user access to relevant Alphagov [teams][]. 

Reviewer: ```suggestion
It’s the individual teams’ responsibility to grant that user access to relevant GDS [GitHub teams][]. 
```

> Once setup is completed, user can access Alphagov resources using the GitHub portal or [SSH connection] (https://help.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh).

Reviewer: ```suggestion
Once setup is completed, staff can access GDS resources on https://github.com or via the command-line.
```

> [GitHub]: https://www.github.com/
> [Alphagov]: https://www.github.com/alphagov/
> [create a new]: https://github.com/join
> [two-form authentication]: https://help.github.com/en/github/authenticating-to-github/configuring-two-factor-authentication
> [teams]: https://github.com/orgs/alphagov/teams
> [SSH connection]: https://help.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh

Reviewer: ```suggestion
[GitHub]: https://www.github.com/
[alphagov]: https://www.github.com/alphagov/
[two-factor authentication]: https://help.github.com/en/github/authenticating-to-github/configuring-two-factor-authentication
[GitHub teams]: https://github.com/orgs/alphagov/teams
```

Reviewer 2: @issyl0 the `[GitHub]` and `[two-factor authentication]` references are unused, did you intend to link to them above?

Reviewer: 👀 Thanks, I edited my suggestions to the intro paragraph (in this awful UI)!

## Access GitHub Alphagov

> People joining GDS do not automatically have access to the GDS GitHub organisation, [alphagov][]. To gain access:

Author: ```suggestion
People joining GDS do not automatically have access to the GDS GitHub organisation, [Alphagov][]. To gain access:
```

> - [Create a GitHub account][GitHub] if you don't have one already. Existing personal accounts are fine to continue using.

Author: ```suggestion
- [Create a GitHub account][] if you don't have one already. Existing personal accounts are fine to continue using.
```

- Enable [two-factor authentication][] on your GitHub account.
- Give your GitHub username to your team's tech lead, or your nearest technical Senior Management Team member.

It’s the individual teams’ responsibility to grant that user access to relevant GDS [GitHub teams][]. 

> Once setup is completed, staff can access GDS resources on https://github.com or via the command-line.

Author: ```suggestion
Once setup is completed, staff can access GDS resources on [GitHub][] or via the command-line.
```

> [GitHub]: https://www.github.com/

Reviewer 2: ```suggestion
[GitHub]: https://github.com/
```

> [Alphagov]: https://www.github.com/alphagov/

Reviewer 2: ```suggestion
[Alphagov]: https://github.com/alphagov/
```

> [create a new]: https://github.com/join

Author: ```suggestion
[Create a GitHub account]: https://github.com/join
```

> [two-form authentication]: https://help.github.com/en/github/authenticating-to-github/configuring-two-factor-authentication

Author: ```suggestion
[two-factor authentication]: https://help.github.com/en/github/authenticating-to-github/configuring-two-factor-authentication
```

> [teams]: https://github.com/orgs/alphagov/teams

Author: ```suggestion
[Github teams]: https://github.com/orgs/alphagov/teams
```

> [Github teams]: https://github.com/orgs/alphagov/teams

Reviewer 2: ```suggestion
[GitHub teams]: https://github.com/orgs/alphagov/teams
```
This doesn't currently match the link above, so this change should fix it

> [SSH connection]: https://help.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh

Reviewer 2: This isn't used and can be removed.

Author: Done
